core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -238,12 +250,20 @@ class GroupMetadataManager(brokerId: Int,
                           consumerId: String,
                           generationId: Int,
                           offsetMetadata: immutable.Map[TopicPartition, OffsetAndMetadata],
-                          responseCallback: immutable.Map[TopicPartition, Errors] => Unit): Option[DelayedStore] = {
+                          responseCallback: immutable.Map[TopicPartition, Errors] => Unit,
+                          producerId: Long = RecordBatch.NO_PRODUCER_ID,
+                          producerEpoch: Short = RecordBatch.NO_PRODUCER_EPOCH): Option[DelayedStore] = {
     // first filter out partitions with offset metadata size exceeding limit
     val filteredOffsetMetadata = offsetMetadata.filter { case (_, offsetAndMetadata) =>
       validateOffsetMetadataLength(offsetAndMetadata.metadata)
     }
 
+    if (!group.hasReceivedConsistentOffsetCommits)
```

Comment:
I think we invoke `prepareStoreOffsets` while holding the group lock, but it might be a good idea to synchronize on the group around this check anyway to make it clear that the group lock is needed.